### PR TITLE
Add waves_per_eu as kernel parameter

### PR DESCRIPTION
**Pull request by oplavsic**
Add ROCM specific kernel parameter waves_per_eu. This parameter will be used to set llvm amd gpu backend attribute "amdgpu-waves-per-eu" which gives compiler hint on how many waves should be executing on single SIMD unit. Default value is 0, and in that case attribute won't be set at all.

### CODE REVIEW

**Review by alefimov-amd — APPROVED** [review]
LGTM,
what about adding some optimization pass on ttg optimization stage, which will automatically compute wavesPerEU and adds it to ttf module?

In this case we do not need to change interface of common `translateTtritonGPUToLLVMTR` and related functions.

File: python/tutorials/06-fused-attention.py [hunk 1]
```diff
@@ -441,9 +441,12 @@ def forward(ctx, q, k, v, causal, sm_scale, split_kernel=False):
         if torch.version.hip is None:
             BLOCK_N = 64 if Lk <= 64 else 32
             num_stages = 4 if Lk <= 64 else 3
+            waves_per_eu = 0 # This is ROCM specific parameter
```
Comment (zhanglx13) [inline python/tutorials/06-fused-attention.py:444]: I think we don't need to expose this variable on the nv path.

**Comment by jayfurmanek** [comment]
This touched a lot of the main function signatures, which has to be carefully considered.
Things out of our control, like jax-triton, call directly into the various compilation routines.

for example:
https://github.com/jax-ml/jax-triton/blob/main/jax_triton/triton_lib.py#L162-L172

We'll also need to understand if/how this affects the new 3rd-party backend set up.

File: python/tutorials/06-fused-attention.py (hunk 1 above)
Comment (zhanglx13) [reply python/tutorials/06-fused-attention.py:444]: I think this part is fine. What Jason brought up about interface changes is more important for now.

**Comment by binarman** [comment]
@jayfurmanek  @zhanglx13 
What do you think about setting this parameter by environment variable?
This way we can keep all public interfaces as-is and nothing breaks.

**Comment by jayfurmanek** [comment]
@oplavsic Can you update this PR? We've made these new parameters optional, so I think we'll go ahead and push this is.

**Comment by oplavsic** [comment]
> @oplavsic Can you update this PR? We've made these new parameters optional, so I think we'll go ahead and push this is.

Done!

**Comment by zhanglx13** [comment]
@jataylo Can you please take a look at this PR to see if it breaks PyTorchIndutor?
Probably yes since we added this `waves-per-eu` to the call_hook. Unfortunately we cannot figure its value at runtime like we did for the `warp_size` parameter. `waves-per-eu` behaves more like `num_warps` and `num_stages`, which are tuning parameters and we "need" different values for different kernels.

**Review by jataylo — APPROVED** [review]
LGTM, tested with PT2.0 and this doesn't cause any issues. 

The issue that was suspected from https://github.com/ROCmSoftwarePlatform/triton/issues/255 does not occur with this change as there is no change to the c_wrapper line in compiler.py https://github.com/ROCmSoftwarePlatform/triton/blob/triton-mlir/python/triton/compiler/compiler.py#L616

**Review by jayfurmanek — APPROVED** [review]

**Pull request merged**